presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -125,6 +127,10 @@ public ExchangeClient(
             Executor pageBufferClientCallbackExecutor)
     {
         this.bufferCapacity = bufferCapacity.toBytes();
+        if (initialRequestCount > 0) {
```

Comment:
Across the codebase we are trying to avoid integers to have a special meaning for `0` or negative values. Let's ban 0. In case we need to disable it, we can always set it to something very large (e.g.: 100000++).